core/deployment/src/test/java/io/quarkiverse/langchain4j/test/AiServiceMethodParametersAnnotationsTest.java

```diff
@@ -0,0 +1,107 @@
+package io.quarkiverse.langchain4j.test;
+
+import static io.restassured.RestAssured.get;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.util.function.Supplier;
+
+import jakarta.validation.constraints.Email;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import dev.langchain4j.data.message.AiMessage;
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import dev.langchain4j.model.output.Response;
+import dev.langchain4j.service.MemoryId;
+import dev.langchain4j.service.UserMessage;
+import io.quarkiverse.langchain4j.RegisterAiService;
+import io.quarkus.test.QuarkusUnitTest;
+
+public class AiServiceMethodParametersAnnotationsTest {
```

Comment:
I don't see how this test tests the new functionality. Can you please explain it? 

Thanks

Reply:
> Is there by any chance a proved way to generate a build item for test purpose only ? That would enhance explainability and maintenance IMHO.

Yes, there is. See [this](https://github.com/quarkusio/quarkus/blob/3.15.0.CR1/extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MongoClientBuildItemConsumerTest.java) example.

Reply:
Ok, the test should be more explicit now.